### PR TITLE
Changed hotcorners retrigger behavior to adjust for faster mouse pointer speed

### PR DESCRIPTION
**Pull request by joshuamars**
Added a check for mouse pointer speed.
Changed the hotcorners retrigger behavior to adjust for faster mouse pointer speed.

This addresses the problems in issues #1334 and (probably) #1330, while still allowing the retrigger behavior to work properly, as it was implemented for #1078.

### CODE REVIEW

File: src/HotCorners/HotCorner.vala
Comment (tintou) [inline src/HotCorners/HotCorner.vala:48]: I think that this part should be in a `construct` block as we will also need to watch for the change of the `speed` property

Comment (tintou) [inline src/HotCorners/HotCorner.vala:48]: Space before starting parenthesis
```suggestion
        mouse_speed_scale_factor = (mouse_settings.get_double ("speed") > 0) ? 
            (1 + mouse_settings.get_double ("speed")) * 2 :
            1;
```

**Review by tintou — DISMISSED** [review]
Thanks for this!
This is indeed a valid approach, just a few things to change.
You can watch for the speed change with:
```vala
mouse_settings.changed["speed"].connect (() => {
    // DO STUFF
});
```

File: src/HotCorners/HotCorner.vala
Comment (joshuamars) [reply src/HotCorners/HotCorner.vala]: Thanks for that! I was wondering how to do that.

**Comment by joshuamars** [comment]
> I wonder whether the issue is the mouse speed or something else. In my case, my mouse speed is also set to 1.0 and I don't experiment this issue:
> 
> ```
> $ gsettings get org.gnome.desktop.peripherals.mouse speed     
> 1.0
> ```
> 
> I think that the issue here are HiDPI screens. @joshuamars are you using any scaling factor?
> 
> image
> 
> If so, a possible approach would be to multiply the pressure by the scaling factor.

My scaling factor is 1.0. I have a larger screen, but not high enough dpi to requiring scaling 🤔. It is 32 US inches, but only 2560x1440

**Comment by JoseExposito** [comment]
Mmm and what's the DPI of your mouse? You can check it online. Let's see if we find what's different in our hardware.

File: src/HotCorners/HotCorner.vala
Comment (JoseExposito) [inline src/HotCorners/HotCorner.vala:41]: Why changing the trigger threshold? This threshold is used the first time the hot corner is activated. With my hardware, it makes difficult to trigger it

File: src/HotCorners/HotCorner.vala
Comment (JoseExposito) [inline src/HotCorners/HotCorner.vala:48]: This threshold is mainly used when using more than one screen. The barrier catches the mouse a bit and allows to move it to the next screen if the user keeps pressing the corner. I don't know if we need to change it.

File: src/HotCorners/HotCorner.vala
Comment (JoseExposito) [inline src/HotCorners/HotCorner.vala:58]: I wonder if just checking for this delay could be a good solution. Do you think it could work?

**Review by JoseExposito — CHANGES_REQUESTED** [review]

File: src/HotCorners/HotCorner.vala
Comment (joshuamars) [reply src/HotCorners/HotCorner.vala:58]: > I wonder if just checking for this delay could be a good solution. Do you think it could work?

I think so. I don't know why I didn't just try that first. I can test it.

File: src/HotCorners/HotCorner.vala
Comment (joshuamars) [reply src/HotCorners/HotCorner.vala:48]: Ok

**Comment by joshuamars** [comment]
> Mmm and what's the DPI of your mouse? You can check it online. Let's see if we find what's different in our hardware.

Just for information, my mouse is Logitech G305, with 12000 DPI

**Comment by JoseExposito** [comment]
> Just for information, my mouse is Logitech G305, with 12000 DPI

Wow, that's a good DPI! I think this is what causes the issue. A non-gaming mouse usually have 1500 ~ 2000 DPI.

FWIW, I believe that exploring the timeout only solution could work.

**Comment by joshuamars** [comment]
> > Just for information, my mouse is Logitech G305, with 12000 DPI
> 
> Wow, that's a good DPI! I think this is what causes the issue. A non-gaming mouse usually have 1500 ~ 2000 DPI.
> 
> FWIW, I believe that exploring the timeout only solution could work.

I think that you are right. I over-complicated it to begin with, and was chasing the wrong rabbit anyways.

Latest commit ([c2d8945](https://github.com/elementary/gala/pull/1421/commits/c2d89459198b7939a2536e73683e0e77db9c895f)) removed everything but the timeout solution

**Review by JoseExposito — APPROVED** [review]
Thanks for the changes! I like this simple approach, LGTM.

**Comment by JoseExposito** [comment]
@tintou I can't merge until you approve the changes, let us know if you like this approach, please. Thanks!

**Pull request merged**
